### PR TITLE
AHRS: Update NED to be relative to origin or home

### PR DESCRIPTION
**Pull request by WickedShell**
This is a fairly large impact commit set, the key point is to decouple the aircraft's concept of relative altitude from home from the EKF's concept of origin. To do this all the get_relative_position(NED) functions have been split into get_relative_position_NED_home or get_relative_position_NED_origin, which are referenced from either the EKF origin or the AHRS home respectively. Every caller library was updated to take advantage of the split.

On the Plane side this then allowed us to always reset the home altitude safely and have a correct and safe altitude referenced to home which is what the user is expecting to see.

One of changes on the plane side that should be emphasized is that plane was converted to cache the value of the relative altitude at the same time we update the current_loc location. Because the relative altitude is defined as relative altitude over home this saves us the math calculations without any loss of accuracy.

The other change (that requires discussion) is that plane was converted to set the home location from the current AHRS location rather then the GPS location. This is intended to address the user seeing the aircraft indicate a different position on the map then the home is set to. If we are considering the AHRS stable enough to fly on, then I think we should accept its assessment of where it is to set home, rather then the GPS location.

This has been flown in plane SITL without a problem. To increase the accuracy of testing the changes there was a noise term applied to the altitude, controlled with SIM_GPS_NOISE which controls the amplitude of a slow sine wave (long term we should allow frequency control as well as amplitude). Items tested where relative missions, absolute missions (and mixing the two). What was not tested was copter SITL, or a real world aircraft.

Things that I would like especially careful review on is the use of location_3d_diff(loc1, loc2) should be returning a NED vector from loc1 to loc2, and that when converting from origin to home references that that math was done correctly.

### CODE REVIEW

File: ArduPlane/commands.cpp [hunk 1]
```diff
@@ -136,17 +136,11 @@ void Plane::update_home()
...
-        Location origin;
-        // if an EKF origin is available then we leave home equal to
-        // the height of that origin. This ensures that our relative
-        // height calculations are using the same origin
-        if (ahrs.get_origin(origin)) {
-            loc.alt = origin.alt;
-        }
+        Location loc;
+        ahrs.get_position(loc);
         ahrs.set_home(loc);
```
Comment (WickedShell) [inline ArduPlane/commands.cpp]: Setting home should only be done if we got a position, will push a fix in a bit.

**Comment by rrr6399** [comment]
Is there logic in the relative altitude calculations to support the case where there are two GPSes that use different altitude datums (WGS84 and MSL),? In other words, is the takeoff altitude for each GPS tracked?

**Comment by WickedShell** [comment]
@rrr6399 You can't mix different datums. Every consumer assumes that its all WGS84. On the todo list (although I haven't made the issue on the list here yet) is that the GPS drivers audit and enforce a WGS84 datum.

**Comment by rrr6399** [comment]
FYI, in 3.4, the default altitude for my primary M8N (?) GPS is MSL and the Reach RTK (M8T) GPS is reporting WGS84. I would think they should be reporting MSL since the terrain is relative to MSL (geoid included).

**Comment by WickedShell** [comment]
EKF assumes WGS84, supporting a different geoid would require a conversation with @priseborough. If the terrain is different then the aircraft is using that should be fixed and a separate issue should be tracking that.

**Comment by WickedShell** [comment]
Updated to account for an error returning cm's rather then meters on NED, and to ensure that altitude relative to home is always provided.

**Comment by WickedShell** [comment]
Travis ran out of time :/

**Comment by OXINARF** [comment]
> Travis ran out of time :/

Passed now :+1:

**Comment by magicrub** [comment]
looks OK, @tridge wants to do one look before it gets merged

**Comment by WickedShell** [comment]
@tridge Any updates?

**Comment by magicrub** [comment]
@tridge ping

**Comment by tridge** [comment]
fantastic piece of work Michael!
pushed to master

**Pull request closed without merging**